Build Failure on Windows

I try to build GAPID on windows 10, but got build error like this:

[60/358] Linking Go executable test\test-core-os-device.exe
FAILED: test/test-core-os-device.exe
cmd.exe /C "cd . && "C:/Program Files/CMake/bin/cmake.exe" -DGO_ENV=C:/Users/zfk/gapid/release/go/goenv.cmake -DBUILDBOT=OFF -DGO_BUILD=test\test-core-os-device.exe -static   -DGO_PACKAGE=github.com/google/gapid/core/os/device -P D:/GoProjects/src/github.com/google/gapid/cmake/Modules/GoCompile.cmake && cmd.exe /C "cd /D D:\GoProjects\src\github.com\google\gapid\core\os\device && "C:\Program Files\CMake\bin\cmake.exe" -DTESTER=C:/Users/zfk/gapid/release/test/test-core-os-device.exe -DGO_PATH="D:/GoProjects/src/github.com/google/gapid/third_party;D:/GoProjects" -P D:/GoProjects/src/github.com/google/gapid/cmake/Modules/GoTest.cmake""
RAN: C:/Users/zfk/gapid/release/test/test-core-os-device.exe
GOT: --- FAIL: TestHost (0.34s)
        testing.go:37: 04:11:43.823 E: Error:Host.Name
                    Got       ``
                    Expect != ``
FAIL

CMake Error at D:/GoProjects/src/github.com/google/gapid/cmake/Modules/GoTest.cmake:25 (message):
  RESULT: 1

I have pulled the latest version, but I still get the following error message:

-- Generating done
-- Build files have been written to: C:/Users/zfk/gapid/release
[474/920] Linking Go executable test\test-core-os-device.exe
FAILED: test/test-core-os-device.exe
cmd.exe /C "cd . && "C:/Program Files/CMake/bin/cmake.exe" -DGO_ENV=C:/Users/zfk/gapid/release/go/goenv.cmake -DBUILDBOT=OFF -DGO_BUILD=test\test-core-os-device.exe -static   -DGO_PACKAGE=github.com/google/gapid/core/os/device -P D:/GoProjects/src/github.com/google/gapid/cmake/Modules/GoCompile.cmake && cmd.exe /C "cd /D D:\GoProjects\src\github.com\google\gapid\core\os\device && "C:\Program Files\CMake\bin\cmake.exe" -DTESTER=C:/Users/zfk/gapid/release/test/test-core-os-device.exe -DGO_PATH="D:/GoProjects/src/github.com/google/gapid/third_party;D:/GoProjects" -P D:/GoProjects/src/github.com/google/gapid/cmake/Modules/GoTest.cmake""
RAN: C:/Users/zfk/gapid/release/test/test-core-os-device.exe
GOT: --- FAIL: TestHost (0.32s)
        testing.go:37: 05:11:31.261 E: Error:Host.Name
                    Got       ``
                    Expect != ``
FAIL

CMake Error at D:/GoProjects/src/github.com/google/gapid/cmake/Modules/GoTest.cmake:25 (message):
  RESULT: 1

And finally I got:
-- LLVMHello ignored -- Loadable modules not supported on this platform.
-- Targeting AArch64
-- Targeting AMDGPU
-- Targeting ARM
-- Targeting BPF
-- Targeting Hexagon
-- Targeting Lanai
-- Targeting Mips
-- Targeting MSP430
-- Targeting NVPTX
-- Targeting PowerPC
-- Targeting RISCV
-- Targeting Sparc
-- Targeting SystemZ
-- Targeting X86
-- Targeting XCore
-- BugpointPasses ignored -- Loadable modules not supported on this platform.
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Users/zfk/gapid/release/llvm/src/llvm-build
ninja: build stopped: subcommand failed.
Error running D:\Downloads\ninja-win\ninja.exe [install]: Process returned error
   Cause: exit status 1
exit status 1

There is no more error messages. I have pulled the code again and found the addtional logging code in my test code:

"snprintf(gContext.mError, sizeof(gContext.mError), \
				     "EGL error: 0x%x when executing:\n   " #x, error); \"

 So I am sure my test version is a new copy. Is there anything wrong with my building operation? I just follow the BUILDING.md guide to setup the environment and run commands in console:(

--- a/core/os/device/deviceinfo/cc/windows/query.cpp
+++ b/core/os/device/deviceinfo/cc/windows/query.cpp
@@ -97,13 +97,12 @@ bool createContext(void*) {
 	}
 	wglMakeCurrent(gContext.mHDC, gContext.mCtx);
 
-    OSVERSIONINFOEX osinfo;
-    gContext.mOsVersion.dwOSVersionInfoSize = sizeof(osinfo);
+    gContext.mOsVersion.dwOSVersionInfoSize = sizeof(gContext.mOsVersion);
     GetVersionEx((OSVERSIONINFO*)(&gContext.mOsVersion));
-    int major = osinfo.dwMajorVersion;
-    int minor = osinfo.dwMinorVersion;
-    int point = osinfo.dwBuildNumber;
-    bool isNTWorkstation = (osinfo.wProductType == VER_NT_WORKSTATION) ? 1 : 0;
+    int major = gContext.mOsVersion.dwMajorVersion;
+    int minor = gContext.mOsVersion.dwMinorVersion;
+    int point = gContext.mOsVersion.dwBuildNumber;
+    bool isNTWorkstation = (gContext.mOsVersion.wProductType == VER_NT_WORKSTATION);
 
 	if (major == 10 && isNTWorkstation) {
 		gContext.mOsName = "Windows 10";
@@ -137,7 +136,7 @@ bool createContext(void*) {
 	GetSystemInfo(&sysInfo);
 	gContext.mNumCores = sysInfo.dwNumberOfProcessors;
 
-	DWORD size = MAX_COMPUTERNAME_LENGTH;
+	DWORD size = MAX_COMPUTERNAME_LENGTH + 1;
 	GetComputerNameA(gContext.mHostName, &size);
 
 	return true;

